tools/tf_graph_tool.py

```diff
@@ -0,0 +1,198 @@
+
+    return inputs, outputs
+
+
+class main(object):
+    @staticmethod
+    def convert_pb_to_pbtxt(input_path, output_path=None):
+        if not output_path:
+            output_path = replace_file_extension(input_path, "pbtxt")
+
+        logging.info("load from %s", input_path)
+        graph_def = load_graph_def_from_pb(input_path)
+
+        logging.info("save to %s", output_path)
+        save_graph_def(graph_def, output_path, as_text=True)
+
+    @staticmethod
+    def convert_pb_to_summary(input_path, output_dir=None, start_tensorboard=False, port=6006):
+        if not output_dir:
+            output_dir = input_path + ".summary"
+
+        logging.info("load from %s", input_path)
+        graph_def = load_graph_def_from_pb(input_path)
+
+        logging.info("save to %s", output_dir)
+        create_directory(output_dir)
+        with tf.Session() as sess:
+            tf.import_graph_def(graph_def, name=get_file_name_without_ext(input_path))
+            train_writer = tf.summary.FileWriter(output_dir)
+            train_writer.add_graph(sess.graph)
+            train_writer.close()
+
+        if start_tensorboard:
+            logging.info("launch tensorboard")
+            os.system("start tensorboard --logdir {} --port {}".format(output_dir, port))
+            os.system("start http://localhost:{}".format(port))
+
+    @staticmethod
+    def get_graph_io_nodes(input_path):
+        logging.info("load from %s", input_path)
+        graph_def = load_graph_def_from_pb(input_path)
+        inputs, outputs = get_graph_def_io_nodes(graph_def)
+        logging.info("graph has:")
+        logging.info("\t%s inputs: %s", len(inputs), ','.join(inputs))
+        logging.info("\t%s (possible) outputs: %s", len(inputs), ','.join(outputs))
+
+    @staticmethod
+    def extract_sub_graph(input_path, output_path=None, dest_nodes=None):
+        if not output_path:
+            output_path = append_file_name_suffix(input_path, "sub")
+
+        logging.info("load from %s", input_path)
+        graph_def = load_graph_def_from_pb(input_path)
+        logging.info("\ttotal node = %s", len(graph_def.node))
+
+        if dest_nodes:
+            dest_nodes = dest_nodes.split(',')
+        else:
+            _, dest_nodes = get_graph_def_io_nodes(graph_def)
+
+        graph_def = graph_util.extract_sub_graph(graph_def, dest_nodes)
+        logging.info("save to %s", output_path)
+        logging.info("\ttotal node = %s", len(graph_def.node))
+        save_graph_def(graph_def, output_path)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers()
+
+    # pb2txt
+    subparser = subparsers.add_parser("pb2txt", help="convert pb to pbtxt")
+    subparser.add_argument("--input", dest="input_path", required=True, help="input pb path")
+    subparser.add_argument("--output", dest="output_path", help="output pbtxt path")
+    subparser.set_defaults(func=main.convert_pb_to_pbtxt)
+
+    # pb2summary
```

Comment:
add some example for using this?

Reply:
sometimes, it is easier to view large graph using tensorboard instead of netron.
netron cannot navigate among nodes by edges easily...

Reply:
i can add something to MD or wiki later.